lib/galaxy_test/api/test_users.py

```diff
@@ -129,6 +134,40 @@ def test_undelete_user(self):
         undeleted_user = self._get(f"users/{user['id']}", admin=True).json()
         assert undeleted_user["deleted"] is False, undeleted_user
 
+    @requires_admin
+    @requires_new_user
+    @requires_new_history
+    @skip_without_tool("cat_data_and_sleep")
+    def test_delete_user_cancel_all_jobs(self):
+        dataset_populator = DatasetPopulator(self.galaxy_interactor)
+        with self._different_user(TEST_USER_EMAIL_DELETE_CANCEL_JOBS):
+            user_id = self._get_current_user_id()
+            history_id = dataset_populator.new_history()
+            hda_id = dataset_populator.new_dataset(history_id)["id"]
+
+            inputs = {
+                "input1": {"src": "hda", "id": hda_id},
+                "sleep_time": 6000,
+            }
+            run_response = dataset_populator.run_tool_raw(
+                "cat_data_and_sleep",
+                inputs,
+                history_id,
+            )
+            self._assert_status_code_is_ok(run_response)
+
+            # Get the job state
+            job_id = run_response.json()["outputs"][0]["id"]
+            job_response = self._get(f"jobs/{job_id}").json()
+            assert job_response["state"] in ["new", "queued", "running"], job_response
+
+            # Delete user will cancel all jobs
+            self._delete(f"users/{user_id}", admin=True)
+
+            # Get the job state again (this time as admin), it should be deleted
+            job_response = self._get(f"jobs/{job_id}", admin=True).json()
```

Comment:
This is going to return all jobs, so will likely not work on real servers. Do you mind moving the test into the integration tests ?

Reply:
> This is going to return all jobs, so will likely not work on real servers.

Sure I can move it to an integration test. But can you elaborate a bit on this? I'm confused, I thought this would return only the job with `job_id` :thinking:

Reply:
Ah, I don't know what I was thinking, that's fine and doesn't require an integration test.